### ISSUE TITLE
Strange behavior of redirect-rules

### ISSUE THREAD

**Issue by dimisa-RUAdList** [uBlockOrigin/uAssets#7822]
**Example, variant 1**
`http://kinozal.tv/`
Google Chrome 84.0.4147.135 
uBO 1.29.2 default + RU AdList

After adding `||piguiqproxy.com/api|$important,xmlhttprequest,redirect=noop.txt` - the site is working fine, the logger shows the following: https://i.imgur.com/DaXMA5o.jpg

**Example, variant 2**
`http://kinozal.tv/`
FF 80.0
uBO 1.29.2 default + RU AdList

After adding `||piguiqproxy.com/api|$important,xmlhttprequest,redirect=noop.txt` - the site is broken (blank page), the logger shows the following: https://i.imgur.com/DaXMA5o.jpg

**Example, variant 3**
`http://kinozal.tv/`
Pale Moon 28.12.0
uBO 1.29.2 default + RU AdList

After adding `||piguiqproxy.com/api|$important,xmlhttprequest,redirect=noop.txt` - the site is broken (blank page), the logger shows the following: https://i.imgur.com/GTwozCm.jpg

**Comment by uBlock-user** [comment]
For case 2, Open the console on Firefox and look for `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at data:text/plain;base64,Cg==. (Reason: CORS request not http).` 

Do you see it ?

**Comment by dimisa-RUAdList** [comment]
Yes I see. Indeed. This is an [old problem](https://bugzilla.mozilla.org/show_bug.cgi?id=1267027) with Firefox. It can be solved by installing [Scriptlet Doctor](https://github.com/JustOff/scriptlet-doctor/releases) (scriptlet-doctor-quantum)
and adding kinozal.tv to the list of domains on which this extension should work. But in the case of uBO Legacy 1.16.4.25 (and scriptlet-doctor-classic), this will no longer help.

**Comment by uBlock-user** [comment]
@dimisa-RUAdList This isn't a CSP issue, see - [#1645683](https://bugzilla.mozilla.org/show_bug.cgi?id=1645683), [#1434357](https://bugzilla.mozilla.org/show_bug.cgi?id=1434357)
